### DISCUSSION TITLE
Disallow custom user input in variable list

### DISCUSSION THREAD

**Discussion by benrubson** [grafana/grafana#38377] [Dashboards, Panels & Visualizations]
Hi,

When defining a _Query_ or _Custom_ variable, values are given in a dropdown list :

![Screenshot 2020-05-18 at 22 22 29](https://user-images.githubusercontent.com/6764151/82256289-57f2f980-9956-11ea-9c5c-a5636cb08504.png)

User can however use this field to enter its own input :

![Screenshot 2020-05-18 at 22 22 44](https://user-images.githubusercontent.com/6764151/82256391-8e307900-9956-11ea-9bbb-ab61413513e8.png)

Which is then accepted and used wherever the variable is used :

![Screenshot 2020-05-18 at 22 22 51](https://user-images.githubusercontent.com/6764151/82256425-9ab4d180-9956-11ea-9362-853a094e47d7.png)

I would like to be able to disallow custom user input in this dropdown list.
So that he can only use the proposed values.
We could still, if relevant / easy to do, allow him to type to search among the proposed values.

As per my findings it's not possible for now, and I did not found any related ticket so far.

Thank you very much 👍

Ben

**Comment by steefenh** [comment]
further more it is possible to put anything you want in the URL Query and the dashboard loads this while the entered variable is not in the drop down list.
Would be good to have a checkbox saying "verify input" for every variable

e.g. you have the following servers in the variable "hosts"

server1
server2
server3

URL says: http://grafana/folder1/dashboard1?orgid=1&hosts=server2 --> OK
URL says: http://grafana/folder1/dashboard1?orgid=1&hosts=server456 --> NOK (do not load data)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
